src/coreclr/vm/ecalllist.h

```diff
@@ -455,6 +448,7 @@ FCFuncEnd()
 
 FCFuncStart(gMethodTableFuncs)
     FCFuncElement("GetNumInstanceFieldBytes", MethodTableNative::GetNumInstanceFieldBytes)
+    FCFuncElement("GetPrimitiveCorElementType", MethodTableNative::GetPrimitiveCorElementType)
```

Comment:
Can it be converted to a QCall?

Reply:
I would keep this as an FCall.

Reply:
Is it because does it would certainly regress performance? Going by the [docs](https://github.com/dotnet/runtime/blob/8169bdee7638502eac41d0d4e04957cbeb103280/docs/design/coreclr/botr/corelib.md):

> In the event the performance of an FCall is required consider creating a QCall and marking it with [SuppressGCTransitionAttribute](https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.suppressgctransitionattribute).

Reply:
Is it because it would certainly regress performance? Going by the [docs](https://github.com/dotnet/runtime/blob/8169bdee7638502eac41d0d4e04957cbeb103280/docs/design/coreclr/botr/corelib.md):

> In the event the performance of an FCall is required consider creating a QCall and marking it with [SuppressGCTransitionAttribute](https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.suppressgctransitionattribute).

Reply:
I would argue for two reasons, yes I am well-versed with the guidance. The first is we already have an existing FCall and adding a QCall create unnecessary asymmetry with little benefit. One could change all of them in this PR but that seems odd. So the second factor would be FCalls are a concern for GC starvation and complexity. In this case there is minimal starvation concern due to how simple/fast it is and no real complexity since it is simply checking a flag on on unmanaged memory.

Changing the group of them seems better than creating a very simple QCall and all the ceremony that comes with it.

Reply:
> it would certainly regress performance?

It would both regress performance a bit and made the code more cumbersome. 

FCalls without HELPER_METHOD_FRAME like this one are ok.

Reply:
I was mostly asking for my own education as in the past when FCalls where replaced by QCalls in e.g. https://github.com/dotnet/runtime/commit/cd7e4cacce3b47a2560bc9ec746f0825a43f99d7, the diff didn't ended up being more cumbersome. Maybe things have changed since? I agree with @AaronRobinsonMSFT to do the class / group wide conversion rather than one-off (focusing on HELPER_METHOD_FRAME ones first).

Reply:
Playing with codegen, calling P/Invoke with `[SuppressGCTransition]` still has slightly higher overhead than calling a method with managed calling convention:

<details>

```csharp
        void Caller()
        {
            GetTickCount();
        }

        [DllImport("kernel32")]
        [SuppressGCTransition]
        static extern int GetTickCount();

        void ControlGroupCaller()
        {
            ControlGroup();
        }

        [MethodImpl(MethodImplOptions.NoInlining)]
        static int ControlGroup() => 23;
```

```asm
; Method CSPlayground.Program:ControlGroupCaller():this (FullOpts)
G_M000_IG01:                ;; offset=0x0000
       sub      rsp, 40

G_M000_IG02:                ;; offset=0x0004
       call     [CSPlayground.Program:ControlGroup():int]
       nop      

G_M000_IG03:                ;; offset=0x000B
       add      rsp, 40
       ret      
; Total bytes of code: 16
```

```asm
; Method CSPlayground.Program:Caller():this (FullOpts)
G_M000_IG01:                ;; offset=0x0000
       push     rbp
       sub      rsp, 32
       lea      rbp, [rsp+0x20]

G_M000_IG02:                ;; offset=0x000A
       mov      rax, 0x7FF8F8830D60

G_M000_IG03:                ;; offset=0x0014
       call     rax ; CSPlayground.Program:GetTickCount():int
       cmp      dword ptr [(reloc 0x7fffa05d305c)], 0
       jne      SHORT G_M000_IG05

G_M000_IG04:                ;; offset=0x001F
       add      rsp, 32
       pop      rbp
       ret      

G_M000_IG05:                ;; offset=0x0025
       call     CORINFO_HELP_POLL_GC
       jmp      SHORT G_M000_IG04
; Total bytes of code: 44
```

</details>

A GC poll is inserted.

Reply:
`SuppressGCTransition` on `QCall` with `GCX_COOP()` etc. on implementation side should elide it in valid cases (which we can't do with a regular `DllImport`).

Reply:
> the diff didn't ended up being more cumbersome.

I think this is a matter of opinion, but I find the QCall effort vs the FCall rather different. For QCalls, (3) things need to happen. First, we need to add a new entry into the QCall list. This can get difficult because of how and where functions are declared. Second, we need to find a place for this and that isn't always clear/obvious when there is already a collection of FCalls. Finally making it all "look natural" can be tricky. The majority of this is style and simple annoyances as opposed to a technical issue. Adding to an existing suite of FCalls can be straight forward, which is usually define/declare it in the same (2) locations and the signature and contents are mostly copy/paste.

The above is only valid in my opinion if the new FCall is *not* introducing a `HELPER_METHOD_FRAME` **and** there are existing FCalls on the type and one is following the pattern.

As far as getting rid of FCalls, I think we should endevour to do them in as bulky a manner as reasonable to review and 
 with respect to the types they are attached to. That means changing 3 FCalls that are unrelated and on different types in a a single PR is odd, but converting 3 FCalls to QCalls or removing them and introducing a new extremely FCall seems reasonable to me. I think the gist here is there is no hard and fast rule, which is why we always suggest contributors to ask about approaches for non-issue related clean-up PRs.